src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiLinkGenerator.cs

```diff
@@ -22,32 +23,83 @@ internal static class OpenApiLinkGenerator
         /// </summary>
         /// <param name="context">The OData context.</param>
         /// <param name="entityType">The Entity type.</param>
-        /// <param name ="sourceElementName">The name of the source of the <see cref="IEdmEntityType" />.</param>
+        /// <param name ="entityName">The name of the source of the <see cref="IEdmEntityType"/> object.</param>
+        /// <param name="entityKind">"The kind of the source of the <see cref="IEdmEntityType"/> object.</param>
+        /// <param name="parameters">"The list of parameters of the incoming operation.</param>
+        /// <param name="navPropOperationId">Optional parameter: The operation id of the source of the NavigationProperty object.</param>
+        /// <param name="targetMultiplicity">"Optional parameter: Flag indicating whether the <see cref="IEdmEntityType"/> object is a collection."</param>
         /// <returns>The created dictionary of <see cref="OpenApiLink"/> object.</returns>
-        public static IDictionary<string, OpenApiLink> CreateLinks(this ODataContext context, IEdmEntityType entityType, string sourceElementName)
+        public static IDictionary<string, OpenApiLink> CreateLinks(this ODataContext context,
+            IEdmEntityType entityType, string entityName, string entityKind,
+            IList<OpenApiParameter> parameters, string navPropOperationId = null,
+            bool targetMultiplicity = false)
         {
-            Utils.CheckArgumentNull(context, nameof(context));
-            Utils.CheckArgumentNull(entityType, nameof(entityType));
-            Utils.CheckArgumentNullOrEmpty(sourceElementName, nameof(sourceElementName));
-
             IDictionary<string, OpenApiLink> links = new Dictionary<string, OpenApiLink>();
-            foreach (IEdmNavigationProperty np in entityType.DeclaredNavigationProperties())
+
+            if (!targetMultiplicity)
```

Comment:
>if (!targetMultiplicity) [](start = 12, length = 24)

if targetMultiplicity == true, this function seems doing nothing?
Why don't you check this boolean out of this function call?

If (!targetMultiplicity)
{
      var links  = context.CreateLinks(...);
}

Reply:
Good catch. Resolved.